packages/nextjs/app/checkinForm/page.tsx

```diff
@@ -1,20 +1,148 @@
 "use client";
 
-import React from "react";
+import React, { SyntheticEvent, useContext, useState } from "react";
+import { useRouter } from "next/navigation";
+import { SchemaEncoder } from "@ethereum-attestation-service/eas-sdk";
 import type { NextPage } from "next";
+import { useAccount } from "wagmi";
+import easConfig from "~~/EAS.config";
+import { EASContext } from "~~/components/EasContextProvider";
+import Mapbox from "~~/components/Mapbox";
 
 // import Link from "next/link";
 
-const CheckinFrom: NextPage = () => {
+const CheckinForm: NextPage = () => {
+  // NextJS redirect
+  const { push } = useRouter();
+
+  const { address: connectedAddress } = useAccount(); //get address from wagmi
+  const nowInSeconds = Math.floor(Date.now() / 1000);
+  const [formValues, setFormValues] = useState({
+    coordinateInputX: 0, // to be picked up by prop
+    coordinateInputY: 0, // to be picked up by prop
+    timestamp: nowInSeconds,
+    data: "Something you wanna add.",
+  });
+
+  // Use EAS SDK
+  const { eas, isReady } = useContext(EASContext);
+  // const [attestation, setAttestation] = useState<Attestation>();
+
+  // Initialize SchemaEncoder with the schema string
+  const schemaEncoder = new SchemaEncoder("string[] coordinates,address subject,uint256 timestamp,bytes32 message");
```

Comment:
Seems important to ensure that schema is structured properly, i.e. using the [JSON Canonicalization Scheme](https://www.rfc-editor.org/rfc/rfc8785)? I say that because with `message` last, we couldn't easily detect that this is the same as another schema

- Key Order: Sort all object keys lexicographically.
- Value Normalization: Remove insignificant whitespace. Represent numbers in a consistent manner. 
- UTF-8 Encoding: Ensure all JSON strings are UTF-8 encoded.

Reply:
We can display key-value pairs in whatever order we want on the UI